startup.py

```diff
@@ -0,0 +1,150 @@
+        Config files to be Maya-specific
+        """
+        if self._synergy_paths is None:
+            # Let the parent class deal with finding the files
+            all_paths = SoftwareLauncher.synergy_paths(self)
+            # Now pick out the specific ones for Maya.
+            self._synergy_paths = [
+                p for p in all_paths if os.path.basename(p).startswith("Maya")
+            ]
+        return self._synergy_paths
+
+    def scan_software(self, versions=None):
+        """
+        Performs a scan for software installations.
+
+        :param versions: List of strings representing versions
+                         to search for. If set to None, search
+                         for all versions. A version string is
+                         DCC-specific but could be something
+                         like "2017", "6.3v7" or "1.2.3.52"
+        :returns: List of :class:`SoftwareVersion` instances
+        """
+        sw_versions = []
+        for synergy_cfg in self.synergy_paths():
+            swv = self._software_version_from_synergy(synergy_cfg)
+            if not swv:
+                self.logger.debug(
+                    "Could not create SoftwareVersion from Synergy config '%s'" %
+                    (synergy_cfg)
+                )
+                continue
+            if versions and isinstance(versions, list) and str(swv.version) not in versions:
+                self.logger.debug(
+                    "Resolved version for Synergy config '%s' [%s] not found "
+                    "in specified list of versions : %s" %
+                    (synergy_cfg, swv.version, versions)
+                )
+                continue
+            sw_versions.append(swv)
+
+        [self._customize_software_version(swv) for swv in sw_versions]
+
+        return sw_versions
+
+    def resolve_software(self, sw_path):
+        """
+        Resolve a software instance for a given DCC path
+
+        :param path:
+        :returns: SoftwareVersion instance
+        """
+        synergy_data = self._synergy_data_from_executable(sw_path)
+        if not synergy_data:
+            return None
+
+        sw_version = self._sofware_version_from_synergy(syn_data=synergy_data)
+        return self._customize_software_version(sw_version)
+
+    def prepare_launch(self, software_version, args, options, file_to_open=None):
+        """
+        Prepares the given software for launch
+
+        :param software_version: Software Version to launch
+        :param args: Command line arguments as strings
+        :param options: DCC specific options to pass
+        :param file_to_open: (Optional) Full path name of a file to open on launch
+        :returns: LaunchInformation instance
+        """
+        clean_env = os.environ.copy()
+        startup_path = os.path.join(self.disk_location, "startup")
+        sgtk.util.append_path_to_env_var("PYTHONPATH", startup_path)
+        os.environ["TANK_ENGINE"] = self.engine_name
```

Comment:
one thought here: do we need to capture the whole env here - couldn't we just return the modifications we want to do, e.g. a dict with `TANK_ENGINE`, `TANK_CONTEXT` etc?